rwengine/src/ai/CharacterController.cpp

```diff
@@ -153,6 +156,222 @@ bool Activities::GoTo::update(CharacterObject *character,
     return false;
 }
 
+bool Activities::DriveTo::update(CharacterObject *character,
```

Comment:
This one is quite long. It looks like there's two main things going on in the function, 1. Find the target position, and 2. steer towards it. These can be their own functions with descriptive names, which means you don't need as many comments.

Reply:
I broke it down into more digestible parts, is it okay now?